<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>research</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <link rel="stylesheet" href="../avenir-white.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<p><a href="https://jessicarick.github.io/testweb">Home</a> | <a href="../cv/cv.html">CV</a> | <a href="../publications/pubs.html">Publications</a> | Research | <a href="../teaching/teaching.html">Teaching</a> | <a href="../software/tools.html">Software</a></p>
<h1 id="about-my-research">About my research</h1>
<h2 id="phylogenetics-and-population-history-of-lates-species-in-lake-tanganyika">Phylogenetics and population history of <em>Lates</em> species in Lake Tanganyika</h2>
<p>In the fall of 2016, I began my PhD research focusing on the genomics of evolution and speciation in fish radiations as a member of the Wagner Lab at the University of Wyoming. My research involves investigating the phylogenetics of endemic Lates (Nile perch) species in Lake Tanganyika, including divergence times, phylogenetic relationships, and population histories, as a method of contextualizing current declines in these fisheries-important fishes. In addition to work on Lates species, I am involved in projects researching Lake Tanganyika dagaa (sardines), Arctic char in northern Alaska, and local Wyoming fish as well.</p>
<h2 id="reference-bias-in-snp-based-phylogenetics">Reference bias in SNP-based phylogenetics</h2>
<p>Analysis of genomic data requires researchers to make a priori decisions on which data to retain and which to filter out, which can introduce biases into the data used in analyses. It is important to understand biases introduced by these data processing steps, which can substantially affect subsequent analyses. There has been little investigation into biases introduced by the choice of which reference genome(s) to use for aligning reads when using reduced-representation or whole genome resequencing data, yet these early steps may substantially alter inferences by systematically biasing which genetic loci are retained after processing. Together with Dr.Â Chad Brock, I am using simulations to investigate the effects of the distance from a reference genome to the in-group taxa (i.e., whether the reference genome is an in- or outgroup individual) and the manner in which this interacts with bioinformatic filtering choices made before analysis even begins in earnest.</p>
</body>
</html>
