<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>smammals-log</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <link rel="stylesheet" href="../../air.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<p><a href="https://jessicarick.github.io/testweb">Home</a> | <a href="../cv/cv.html">CV</a> | <a href="../publications/pubs.html">Publications</a> | <a href="../research/research.html">Research</a> | <a href="../teaching/teaching.html">Teaching</a> | <a href="../software/tools.html">Software</a> | <a href="../projects/projects.html">Projects</a></p>
<h1 id="uhuru-small-mammal-genetics">UHURU Small Mammal Genetics</h1>
<h2 id="log-of-methods">Log of Methods</h2>
<center>
<a href="#recent">jump to most recent</a>
</center>
<h3 id="july-2020">July 2020</h3>
<h4 id="starting-over-now-with-more-data">Starting over – now with more data</h4>
<h5 id="step-1-barcode-parsing">Step 1: Barcode parsing</h5>
<ul>
<li>using old barcode script for 1,2,3; new barcode script for 4,5,6<br />
</li>
<li>took &lt; 2 days for 1,2,3; timeout for 4,5,6<br />
</li>
<li>took 2 days, 20 hours for 4,6<br />
</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="co"># UHURU1</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="ex">Good</span> mids count: 216880875</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="ex">Bad</span> mids count: 10370274 (4.8%)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: 18036770</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: 3703</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a><span class="co"># UHURU2</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a><span class="ex">Good</span> mids count: 208777637</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a><span class="ex">Bad</span> mids count: 9467255 (4.5%)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: 18856287</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: 5192</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true"></a><span class="co"># UHURU3</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true"></a><span class="ex">Good</span> mids count: 213651963</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true"></a><span class="ex">Bad</span> mids count: 9623340 (4.5%)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: 19246222</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: 3445</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true"></a><span class="co"># UHURU4 </span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true"></a><span class="ex">Good</span> mids count: 176225154</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true"></a><span class="ex">Bad</span> mids count: 26225207 (12.9%)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: 229962</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: 783</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true"></a><span class="co"># UHURU5 </span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true"></a><span class="ex">Good</span> mids count: </span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true"></a><span class="ex">Bad</span> mids count:</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: </span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: </span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true"></a><span class="co"># UHURU6</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true"></a><span class="ex">Good</span> mids count: 199599451</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true"></a><span class="ex">Bad</span> mids count: 29505980 (12.9%)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true"></a><span class="ex">Number</span> of seqs with potential MSE adapter in seq: 239013</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true"></a><span class="ex">Seqs</span> that were too short after removing MSE and beyond: 858</span></code></pre></div>
<h3 id="august-2020">August 2020</h3>
<p>#####Step 2: Barcode trimming</p>
<p>Now that I finnnnnnallllly have all of the barcodes parsed, split, and concatenated (from all the different libraries), I need to trim the reads down to 85bp to make Stacks happy. This time, I’m going to use a combo of <code>fastx_toolkit</code> and <code>seqkit</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="ex">module</span> load fastx_toolkit</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a><span class="ex">module</span> load seqkit</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a><span class="ex">module</span> load parallel</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true"></a><span class="ex">parallel</span> -j 16 fastx_trimmer -f 1 -l 85 -Q33 -i {} <span class="ex">-o</span> <span class="dt">{.}</span>.trim.fastq ::: *[0-9].fastq</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true"></a><span class="ex">parallel</span> seqkit seq {} <span class="ex">-m</span> 85 -o ../../04-all_samples/<span class="dt">{.}</span>.trim.fq.gz ::: *[0-9].trim.fastq</span></code></pre></div>
<p>Then, I moved each trimmed fastq into the appropriate species’ directory. Some files are throwing an error– maybe something went wrong with the splitting/concatenating of these? <em>I’ll have to figure out which they are and investigate.</em></p>
<p>#####Step 3: Stacks workflow</p>
<p>From here, I used the <code>run_stacks.sh</code> script for each population, which runs (in sequence) <code>ustacks</code>, <code>cstacks</code>, <code>sstacks</code>, <code>tsv2bam</code>, <code>gstacks</code>, and <code>populations</code>. Results from <code>populations</code> module:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="co">## TAHA</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="ex">Removed</span> 171593 loci that did not pass sample/population constraints from 375129 loci.</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a><span class="ex">Kept</span> 203536 loci, composed of 17316679 sites<span class="kw">;</span> <span class="ex">1044</span> of those sites were filtered, 121655 variant sites remained.</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.08bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true"></a>  <span class="ex">C</span>: 4.2107 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.32013<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 17316622/121655/91732<span class="kw">;</span> <span class="ex">private</span> alleles: 43921</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true"></a>  <span class="ex">N</span>: 2.4155 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.30441<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 17316622/121655/69204<span class="kw">;</span> <span class="ex">private</span> alleles: 21393</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true"></a><span class="co">## ELRU</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true"></a><span class="ex">Genotyped</span> 771664 loci:</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true"></a>  <span class="ex">effective</span> per-sample coverage: mean=8.5x, stdev=2.0x, min=5.9x, max=13.8x</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true"></a>  <span class="ex">mean</span> number of sites per locus: 85.0</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true"></a>  <span class="ex">a</span> consistent phasing was found for 165507 of out 189195 (87.5%) <span class="ex">diploid</span> loci needing phasing</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true"></a><span class="ex">Removed</span> 381300 loci that did not pass sample/population constraints from 771664 loci.</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true"></a><span class="ex">Kept</span> 390364 loci, composed of 33216681 sites<span class="kw">;</span> <span class="ex">2245</span> of those sites were filtered, 406799 variant sites remained.</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.09bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true"></a>  <span class="ex">C</span>: 4.7968 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.29472<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 25852749/346427/244377<span class="kw">;</span> <span class="ex">private</span> alleles: 70368</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true"></a>  <span class="ex">N</span>: 2.663 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.26922<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 30770026/384475/199979<span class="kw">;</span> <span class="ex">private</span> alleles: 55363</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true"></a>  <span class="ex">S</span>: 3.1429 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.27721<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 31115608/385001/214027<span class="kw">;</span> <span class="ex">private</span> alleles: 55145</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true"></a><span class="co">## AEHI</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true"></a></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true"></a><span class="ex">Genotyped</span> 924553 loci:</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true"></a>  <span class="ex">effective</span> per-sample coverage: mean=35.2x, stdev=38.0x, min=6.4x, max=256.8x</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true"></a>  <span class="ex">mean</span> number of sites per locus: 85.0</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true"></a>  <span class="ex">a</span> consistent phasing was found for 39999 of out 53646 (74.6%) <span class="ex">diploid</span> loci needing phasing</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true"></a><span class="ex">Removed</span> 866065 loci that did not pass sample/population constraints from 924553 loci.</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true"></a><span class="ex">Kept</span> 58488 loci, composed of 4980048 sites<span class="kw">;</span> <span class="ex">1994</span> of those sites were filtered, 56485 variant sites remained.</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.15bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true"></a></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true"></a>  <span class="ex">C</span>: 3.531 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.24091<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 4855649/54115/28047<span class="kw">;</span> <span class="ex">private</span> alleles: 4868</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true"></a>  <span class="ex">N</span>: 5.9015 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.21214<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 1162507/22752/13125<span class="kw">;</span> <span class="ex">private</span> alleles: 3087</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true"></a>  <span class="ex">S</span>: 12.966 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.26049<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 4723322/53624/47180<span class="kw">;</span> <span class="ex">private</span> alleles: 21358</span></code></pre></div>
<p>It seems that some of these didn’t work out very well, so I’m going to go back through and remove individuals that have low coverage (&lt; 50000 reads to start) and re-run the <code>cstacks</code> through <code>populations</code> modules. For each species, I did the following:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="kw">for</span> <span class="fu">file</span> in *.fq.gz</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a>    <span class="kw">do</span> <span class="bu">echo</span> <span class="va">$file</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a>    <span class="va">reads=</span><span class="kw">`</span><span class="fu">zgrep</span> <span class="st">&#39;^@&#39;</span> <span class="va">$file</span> <span class="kw">|</span> <span class="fu">wc</span> -l<span class="kw">`</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a>    <span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$file</span><span class="st"> </span><span class="va">$reads</span><span class="st">&quot;</span> <span class="op">&gt;&gt;</span> ../reads_per_ind_AEHI.txt</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a><span class="kw">done</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a><span class="ex">module</span> load r</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a><span class="ex">R</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a>AEHI &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;reads_per_ind_AEHI.txt&quot;</span>, <span class="dt">header=</span>F)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a><span class="kw">colnames</span>(AEHI) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ind&quot;</span>,<span class="st">&quot;reads&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a><span class="kw">summary</span>(AEHI<span class="op">$</span>reads)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a><span class="kw">length</span>(AEHI<span class="op">$</span>ind[AEHI<span class="op">$</span>reads <span class="op">&lt;</span><span class="st"> </span><span class="dv">20000</span>])</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a><span class="kw">write.table</span>(AEHI<span class="op">$</span>ind[AEHI<span class="op">$</span>reads <span class="op">&lt;</span><span class="st"> </span><span class="dv">20000</span>], <span class="st">&quot;lowcov_AEHI&quot;</span>, </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a>    <span class="dt">col.names=</span>F, <span class="dt">row.names=</span>F, <span class="dt">quote=</span>F)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a><span class="co">## AEHI</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true"></a> Min. 1st Qu.  Median    Mean 3rd Qu.    Max.</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true"></a>      <span class="dv">1</span>  <span class="dv">103276</span>  <span class="dv">528904</span>  <span class="dv">740004</span> <span class="dv">1133816</span> <span class="dv">4534199</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true"></a> <span class="dv">15</span> <span class="co"># less than 20k reads</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true"></a> <span class="co">## ELRU</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true"></a>    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true"></a> <span class="dv">117551</span> <span class="dv">1222813</span> <span class="dv">1582039</span> <span class="dv">1708879</span> <span class="dv">1965295</span> <span class="dv">5652582</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true"></a> <span class="dv">0</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true"></a><span class="co">## SAME</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true"></a>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true"></a>   <span class="dv">1093</span> <span class="dv">1080934</span> <span class="dv">2074135</span> <span class="dv">2074894</span> <span class="dv">3174265</span> <span class="dv">4726248</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true"></a> <span class="dv">3</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true"></a><span class="co">## GERO</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true"></a>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true"></a>   <span class="dv">1091</span>  <span class="dv">383796</span> <span class="dv">1071208</span> <span class="dv">1348931</span> <span class="dv">2071609</span> <span class="dv">5858744</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true"></a> <span class="dv">8</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true"></a><span class="co">## TAHA</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true"></a>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true"></a>    <span class="dv">165</span>  <span class="dv">966798</span> <span class="dv">1318086</span> <span class="dv">1436825</span> <span class="dv">1963732</span> <span class="dv">4086397</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true"></a> <span class="dv">4</span></span></code></pre></div>
<p>After removing individuals with fewer than 50k reads, I ran the sstacks-populations steps for all four species again.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a><span class="co"># AEHI</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a><span class="ex">Removed</span> 836692 loci that did not pass sample/population constraints from 920951 loci.</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a><span class="ex">Kept</span> 84259 loci, composed of 7173027 sites<span class="kw">;</span> <span class="ex">2006</span> of those sites were filtered, 73621 variant sites remained.</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.13bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true"></a>  <span class="ex">C</span>: 3.3548 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.24952<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 6711468/68129/35594<span class="kw">;</span> <span class="ex">private</span> alleles: 6994</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true"></a>  <span class="ex">N</span>: 4.6018 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.23599<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 3063727/40707/22648<span class="kw">;</span> <span class="ex">private</span> alleles: 5426</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true"></a>  <span class="ex">S</span>: 11.78 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.26676<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 6448224/67514/58585<span class="kw">;</span> <span class="ex">private</span> alleles: 25538</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true"></a><span class="co"># TAHA</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true"></a></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true"></a><span class="ex">Removed</span> 139783 loci that did not pass sample/population constraints from 332696 loci.</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true"></a><span class="ex">Kept</span> 192913 loci, composed of 16410871 sites<span class="kw">;</span> <span class="ex">928</span> of those sites were filtered, 99125 variant sites remained.</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.07bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true"></a>  <span class="ex">N</span>: 2.4911 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.30759<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 16410825/99125/57910<span class="kw">;</span> <span class="ex">private</span> alleles: 14495</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true"></a>  <span class="ex">C</span>: 4.7358 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.32593<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 16410825/99125/78004<span class="kw">;</span> <span class="ex">private</span> alleles: 34589</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true"></a></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true"></a><span class="co"># SAME</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true"></a><span class="ex">Removed</span> 921240 loci that did not pass sample/population constraints from 1284600 loci.</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true"></a><span class="ex">Kept</span> 363360 loci, composed of 30926106 sites<span class="kw">;</span> <span class="ex">2532</span> of those sites were filtered, 273227 variant sites remained.</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.11bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true"></a>  <span class="ex">C</span>: 7.2428 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.22604<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 28949904/257672/161307<span class="kw">;</span> <span class="ex">private</span> alleles: 24344</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true"></a>  <span class="ex">N</span>: 4.0304 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.21547<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 25558538/233723/114424<span class="kw">;</span> <span class="ex">private</span> alleles: 13621</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true"></a>  <span class="ex">S</span>: 15.609 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.2462<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 30178865/267872/229620<span class="kw">;</span> <span class="ex">private</span> alleles: 74261</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true"></a></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true"></a><span class="co"># GERO</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true"></a></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true"></a><span class="ex">Removed</span> 948377 loci that did not pass sample/population constraints from 1227935 loci.</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true"></a><span class="ex">Kept</span> 279558 loci, composed of 23820166 sites<span class="kw">;</span> <span class="ex">2677</span> of those sites were filtered, 275714 variant sites remained.</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.21bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true"></a></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true"></a>  <span class="ex">C</span>: 15.161 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.21688<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 21901957/257043/194755<span class="kw">;</span> <span class="ex">private</span> alleles: 29552</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true"></a>  <span class="ex">N</span>: 10.667 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.21309<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 22238852/260269/173990<span class="kw">;</span> <span class="ex">private</span> alleles: 21975</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true"></a>  <span class="ex">S</span>: 18.172 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.21668<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 22502712/264402/206734<span class="kw">;</span> <span class="ex">private</span> alleles: 37665</span></code></pre></div>
<h3 id="recent">August 18, 2020</h3>
<p>Now, I’m working with AEHI and GERO to try and improve the Stacks results. After filtering in R, AEHI was left with 54 individuals (out of 9,286) and 9,286 SNPs; GERO had 149 individuals (out of 220) and only 5,097 SNPs. I changed the parameters in <code>ustacks</code> to <code>-m 3 -M 5 -N 7 --model_type bounded --bound_high 0.05</code> based off of some recommendations that I’ve seen– we’ll see how it goes! I also made it so that all individuals belong to the same population, instead of specifying C/N/S, and changed to <code>-p 1 -r 0.1</code>.</p>
<p>Here are the results from <code>populations</code>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a><span class="co"># AEHI</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true"></a><span class="ex">Removed</span> 1229518 loci that did not pass sample/population constraints from 1279334 loci.</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true"></a><span class="ex">Kept</span> 49816 loci, composed of 4242349 sites<span class="kw">;</span> <span class="ex">2272</span> of those sites were filtered, 60891 variant sites remained.</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true"></a><span class="ex">Mean</span> genotyped sites per locus: 85.16bp (stderr 0.00)<span class="ex">.</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true"></a><span class="ex">Population</span> summary statistics (more detail in populations.sumstats_summary.tsv)<span class="bu">:</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true"></a>  <span class="ex">AEHI</span>: 30.408 samples per locus<span class="kw">;</span> <span class="ex">pi</span>: 0.23675<span class="kw">;</span> <span class="ex">all/variant/polymorphic</span> sites: 4242341/60891/60891<span class="kw">;</span> <span class="ex">private</span> alleles: 0</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true"></a><span class="co"># GERO</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true"></a></span></code></pre></div>
</body>
</html>
